### DISCUSSION TITLE
v0.2.36

### DISCUSSION THREAD

**Discussion by jackgerrits** [microsoft/autogen#3595] [Announcements]
> [!IMPORTANT]
> In order to better align with a new multi-packaging structure we have coming very soon, AutoGen is now available on PyPi as [`autogen-agentchat`](https://pypi.org/project/autogen-agentchat/) as of version `0.2.36`.

## Highlights

* Ability to add MessageTransforms to the GroupChat's Select Speaker nested chat (speaker_selection_method='auto') by @marklysze in https://github.com/microsoft/autogen/pull/2719
* Integrate Mem0 for providing long-term memory for AI Agents by @Dev-Khant in https://github.com/microsoft/autogen/pull/3389
* Amazon Bedrock Client for AutoGen by @Hk669 in https://github.com/microsoft/autogen/pull/3232
* Add Couchbase VectorDB Support by @lokesh-couchbase in https://github.com/microsoft/autogen/pull/3525
* Portkey Integration with Autogen by @siddharthsambharia-portkey in https://github.com/microsoft/autogen/pull/3395
* Add Cerebras Integration by @henrytwo in https://github.com/microsoft/autogen/pull/3585
* Ollama Client (with tool calling) by @marklysze in https://github.com/microsoft/autogen/pull/3056
* Ollama library v0.3.3 update with changes to options handling by @marklysze in https://github.com/microsoft/autogen/pull/3593

## What's Changed
* [.Net] Dispose kernel after running dotnet interactive tests by @LittleLittleCloud in https://github.com/microsoft/autogen/pull/3378
* Ensure 'name' on initial message by @marklysze in https://github.com/microsoft/autogen/pull/2635
* [.Net] Enable step-by-step execution for two-agent conversation SendAsync API by @LittleLittleCloud in https://github.com/microsoft/autogen/pull/3360
* Add contributor list via contributors.md by @ekzhu in https://github.com/microsoft/autogen/pull/3384
* Update CONTRIBUTORS.md; fix cell order by @gagb in https://github.com/microsoft/autogen/pull/3386
* Update CONTRIBUTORS.md; fix broken URL by @gagb in https://github.com/microsoft/autogen/pull/3387
* Update AutoGen.Net Contributors.md by @LittleLittleCloud in https://github.com/microsoft/autogen/pull/3393
* Update CONTRIBUTORS.md by @rickyloynd-microsoft in https://github.com/microsoft/autogen/pull/3391
* Add Language Agent Tree Search (LATS) notebook by @andyz245 in https://github.com/microsoft/autogen/pull/3376
* [.Net] Release 0.1.0 by @LittleLittleCloud in https://github.com/microsoft/autogen/pull/3398
* [.Net] Rename AutoGen.OpenAI to AutoGen.OpenAI.V1 by @LittleLittleCloud in https://github.com/microsoft/autogen/pull/3358
* Update Docker.md；fix broken URL by @New-World-2019 in https://github.com/microsoft/autogen/pull/3399
* Fix embedding function assignment in QdrantVectorDB by @eddyfidel in https://github.com/microsoft/autogen/pull/3396
* Add mongodb to topic guide docs by @thinkall in https://github.com/microsoft/autogen/pull/3400
* Ability to add MessageTransforms to the GroupChat's Select Speaker nested chat (speaker_selection_method='auto') by @marklysze in https://github.com/microsoft/autogen/pull/2719
* Fix for group chat resume - full chat history for each agent by @marklysze in https://github.com/microsoft/autogen/pull/3412
* Update agent_chat.md；Fix broken URL by @New-World-2019 in https://github.com/microsoft/autogen/pull/3416
* Add None back to remove_termination_string for a_resume in group chat by @heyitsaamir in https://github.com/microsoft/autogen/pull/3410
* Amazon Bedrock Client for AutoGen by @Hk669 in https://github.com/microsoft/autogen/pull/3232
* fix `ImportFromModule` is unhashable issue by @zcipod in https://github.com/microsoft/autogen/pull/3362
* Transform to add an agent's name into the message content by @marklysze in https://github.com/microsoft/autogen/pull/3334
* [Gallery] Adding a new Finance Tracker Demo Application by @Kirushikesh in https://github.com/microsoft/autogen/pull/3414
* Update contributors by @davorrunje in https://github.com/microsoft/autogen/pull/3420
* Update privacy link in readme and add Consumer Health Privacy notice on website by @jackgerrits in https://github.com/microsoft/autogen/pull/3422
* Add studio pre-print by @gagb in https://github.com/microsoft/autogen/pull/3423
* Update README.md by @gagb in https://github.com/microsoft/autogen/pull/3424
* Update CITATION.cff by @gagb in https://github.com/microsoft/autogen/pull/3427
* Add details about GAIA benchmark evaluation by @jackgerrits in https://github.com/microsoft/autogen/pull/3433
* Add missing contributors by @gagb in https://github.com/microsoft/autogen/pull/3426
* Update package.json, remove gh-pages dep by @victordibia in https://github.com/microsoft/autogen/pull/3435
* [.Net] Add AutoGen.OpenAI package that uses OpenAI v2 SDK by @LittleLittleCloud in https://github.com/microsoft/autogen/pull/3402
* Autogenstudio Update - Support for Anthropic/Mistral, Other Updates by @victordibia in https://github.com/microsoft/autogen/pull/3439
* Fix AutoGen Studio pathPrefix error on Windows by @New-World-2019 in https://github.com/microsoft/autogen/pull/3406
* fix: tool calling cohere by @Anirudh31415926535 in https://github.com/microsoft/autogen/pull/3355
* [.Net] add output schema to generateReplyOption by @LittleLittleCloud in https://github.com/microsoft/autogen/pull/3450
* Use utf-8 encoding at subprocess.run() by @giorgossideris in https://github.com/microsoft/autogen/pull/3454
* fix model_config SQLModel by @Hk669 in https://github.com/microsoft/autogen/pull/3457
* Fix for Anthropic client class so system messages aren't lost by @marklysze in https://github.com/microsoft/autogen/pull/3392
* docs: fix autogen install url by @Lulalaby in https://github.com/microsoft/autogen/pull/3459
* no more then -> no more than #typo by @scenaristeur in https://github.com/microsoft/autogen/pull/3469
* [Graph RAG] Init Commit with GraphRag interfaces by @randombet in https://github.com/microsoft/autogen/pull/3388
* Update usage of built-in LLMs in Microsoft Fabric by @thinkall in https://github.com/microsoft/autogen/pull/3474
* [.Net] In AutoGen.OpenAI and AutoGen.OpenAI.V1, stop setting name field when assistant message contains tool call by @LittleLittleCloud in https://github.com/microsoft/autogen/pull/3481
* [.Net] release v0.2.0 by @LittleLittleCloud in https://github.com/microsoft/autogen/pull/3483
* Integrate Mem0 for providing long-term memory for AI Agents by @Dev-Khant in https://github.com/microsoft/autogen/pull/3389
* Update readme by @gagb in https://github.com/microsoft/autogen/pull/3485
* Ecosystem Mem0: change title by @Dev-Khant in https://github.com/microsoft/autogen/pull/3492
* Support for Human Input Mode in AutoGen Studio by @victordibia in https://github.com/microsoft/autogen/pull/3484
* Fix the bug of creating a new session on the AutoGen Studio playgroun… by @New-World-2019 in https://github.com/microsoft/autogen/pull/3407
* Portkey Integration with Autogen by @siddharthsambharia-portkey in https://github.com/microsoft/autogen/pull/3395
* Update pyproject.toml for AutoGen Studio by @gagb in https://github.com/microsoft/autogen/pull/3495
* Update README.md by @gagb in https://github.com/microsoft/autogen/pull/3499
* [.Net][AutoGen.OpenAI] Allow nullable system message, temperature and max token to support o1-preview model by @LittleLittleCloud in https://github.com/microsoft/autogen/pull/3524
* [.NET] Release v0.2.1 by @DavidLuong98 in https://github.com/microsoft/autogen/pull/3529
* Fix issue caused by chromadb error type change by @thinkall in https://github.com/microsoft/autogen/pull/3557
* Bump express from 4.19.2 to 4.21.0 in /website by @dependabot in https://github.com/microsoft/autogen/pull/3552
* typo intiate_chat -> initiate_chat by @scenaristeur in https://github.com/microsoft/autogen/pull/3533
* #3538: issue: 'https://docs.trychroma.com/embeddings' hyperlink is … by @CellCS in https://github.com/microsoft/autogen/pull/3539
* Fix syntax error in user-defined-functions docs (unecessarily capital… by @SuchitG04 in https://github.com/microsoft/autogen/pull/3336
* Bump webpack from 5.89.0 to 5.94.0 in /website by @dependabot in https://github.com/microsoft/autogen/pull/3438
* Update LLM Caching link in Optional-Dependencies.md by @evanhadfield in https://github.com/microsoft/autogen/pull/2885
* Fix for async group chat resume - full chat history for each agent by @hardchor in https://github.com/microsoft/autogen/pull/3558
* fix: update cohere tool calling multi agents by @Anirudh31415926535 in https://github.com/microsoft/autogen/pull/3488
* Fix role mapping in GPTAssistantAgent for OpenAI API compatibility by @evandavid1 in https://github.com/microsoft/autogen/pull/3555
* Fix Exception in a_generate_reply when function_call is None by @artero in https://github.com/microsoft/autogen/pull/3250
* Catch token count issue while streaming with customized models by @BeibinLi in https://github.com/microsoft/autogen/pull/3241
* Fix AttributeError: 'NoneType' object has no attribute 'name' by @drewgillson in https://github.com/microsoft/autogen/pull/3127
* Add cookies from http session to web socket used by JupyterCodeExecutor by @timrbula in https://github.com/microsoft/autogen/pull/3312
* Bump micromatch from 4.0.5 to 4.0.8 in /website by @dependabot in https://github.com/microsoft/autogen/pull/3417
* Throttle API Calls by @WaelKarkoub in https://github.com/microsoft/autogen/pull/2283
* Add kwargs to pass to docker container that runs the Jupyter Server by @rahulAira in https://github.com/microsoft/autogen/pull/3028
* Remove api key validation by @jackgerrits in https://github.com/microsoft/autogen/pull/3569
* Add init to agent_eval by @jackgerrits in https://github.com/microsoft/autogen/pull/3568
* Add Couchbase VectorDB Support by @lokesh-couchbase in https://github.com/microsoft/autogen/pull/3525
* WebSurfer Updated (Selenium, Playwright, and support for many filetypes) by @afourney in https://github.com/microsoft/autogen/pull/1929
* Embedding function should always return a list of a list of vectors by @paubins2 in https://github.com/microsoft/autogen/pull/3570
* refactor: use Qdrant's Query API by @Anush008 in https://github.com/microsoft/autogen/pull/3428
* Removes EasyOCR dependency from mdconvert. by @afourney in https://github.com/microsoft/autogen/pull/3573
* Deploy website using action instead of branch by @jackgerrits in https://github.com/microsoft/autogen/pull/3578
* Update token permissions for site deployment by @jackgerrits in https://github.com/microsoft/autogen/pull/3580
* Remove announcement bar by @jackgerrits in https://github.com/microsoft/autogen/pull/3579
* Added instructions to launch frontend server when running from source by @EnGassa in https://github.com/microsoft/autogen/pull/3517
* async user hook support added by @robraux in https://github.com/microsoft/autogen/pull/3583
* Fix line endings of test/browser_utils/test_files/test_serp.html by @jackgerrits in https://github.com/microsoft/autogen/pull/3586
* Add Cerebras Integration by @henrytwo in https://github.com/microsoft/autogen/pull/3585
* Add AI email agent using Composio to notebook by @cagnusmarlsen in https://github.com/microsoft/autogen/pull/3567
* Remove API key warning by @jackgerrits in https://github.com/microsoft/autogen/pull/3589
* Disable Azure tests and fix open ai tests by @ekzhu in https://github.com/microsoft/autogen/pull/3587
* Ollama Client (with tool calling) by @marklysze in https://github.com/microsoft/autogen/pull/3056
* Ollama library v0.3.3 update with changes to options handling by @marklysze in https://github.com/microsoft/autogen/pull/3593
* Use native logger formatter by @ekzhu in https://github.com/microsoft/autogen/pull/3594

## New Contributors
* @andyz245 made their first contribution in https://github.com/microsoft/autogen/pull/3376
* @New-World-2019 made their first contribution in https://github.com/microsoft/autogen/pull/3399
* @eddyfidel made their first contribution in https://github.com/microsoft/autogen/pull/3396
* @zcipod made their first contribution in https://github.com/microsoft/autogen/pull/3362
* @Kirushikesh made their first contribution in https://github.com/microsoft/autogen/pull/3414
* @Anirudh31415926535 made their first contribution in https://github.com/microsoft/autogen/pull/3355
* @Lulalaby made their first contribution in https://github.com/microsoft/autogen/pull/3459
* @scenaristeur made their first contribution in https://github.com/microsoft/autogen/pull/3469
* @Dev-Khant made their first contribution in https://github.com/microsoft/autogen/pull/3389
* @siddharthsambharia-portkey made their first contribution in https://github.com/microsoft/autogen/pull/3395
* @SuchitG04 made their first contribution in https://github.com/microsoft/autogen/pull/3336
* @evanhadfield made their first contribution in https://github.com/microsoft/autogen/pull/2885
* @hardchor made their first contribution in https://github.com/microsoft/autogen/pull/3558
* @evandavid1 made their first contribution in https://github.com/microsoft/autogen/pull/3555
* @artero made their first contribution in https://github.com/microsoft/autogen/pull/3250
* @drewgillson made their first contribution in https://github.com/microsoft/autogen/pull/3127
* @timrbula made their first contribution in https://github.com/microsoft/autogen/pull/3312
* @rahulAira made their first contribution in https://github.com/microsoft/autogen/pull/3028
* @lokesh-couchbase made their first contribution in https://github.com/microsoft/autogen/pull/3525
* @paubins2 made their first contribution in https://github.com/microsoft/autogen/pull/3570
* @EnGassa made their first contribution in https://github.com/microsoft/autogen/pull/3517
* @henrytwo made their first contribution in https://github.com/microsoft/autogen/pull/3585
* @cagnusmarlsen made their first contribution in https://github.com/microsoft/autogen/pull/3567

**Full Changelog**: https://github.com/microsoft/autogen/compare/v0.2.35...v0.2.36

<hr /><em>This discussion was created from the release <a href='https://github.com/microsoft/autogen/releases/tag/v0.2.36'>v0.2.36</a>.</em>

**Comment by robraux** [comment]
Can someone please explain, or link to a discussion or explanation around the apparent fork?

https://github.com/autogen-ai/autogen vs. https://github.com/microsoft/autogen (this repo)

Each have different yet similar release versions, along with very similar python package naming. This is confusing.

**Reply by jackgerrits** [reply]
Understood about the confusion. There are many forks of the AutoGen repo created by people for different reasons. This ([Microsoft/autogen](https://github.com/microsoft/autogen)) is the original repo for AutoGen project, which is supported and maintained by Microsoft.
